homeassistant/components/image_processing/__init__.py

```diff
@@ -71,25 +71,24 @@
 def scan(hass, entity_id=None):
     """Force process an image."""
     data = {ATTR_ENTITY_ID: entity_id} if entity_id else None
-    hass.services.call(DOMAIN, SERVICE_SCAN, data)
+    hass.async_add_job(hass.services.async_call(
```

Comment:
`scan` isn't expected to be called from an async context so we can't use `async_add_job` here. Why should we change the code here? Probably instead add an `async_scan` function if it's needed.

Reply:
Well, scan isn't currently called anywhere.

And this rewrite is based on the camera component "enable_motion_detection", "disable_motion_detection". 

So I just followed the pattern to improve consistence cross components.

Reply:
Those are not named correctly either. Home assistant uses `async_` to mark function names that should be called from an async context. We should stick to that.

Reply:
remove it, when no code point to this

Reply:
Actually the all purpose of this is to use it :)